### PR TITLE
refactor(react-drawer): rename main DrawerOverlay/DrawerInline components

### LINKED ISSUE #29238: Doc: typos and re-wording suggestions
A list of typos in the docs and suggestions for re-wording parts. Up to you if you want to re-word things 🙂

---

- [ ] DrawerOverlay description

> DrawerOverlay: Represents an overlay Drawer. This component renders on top of the whole page. By default blocks the screen and will require users full attention. Uses Dialog component under the hood.

```suggestion
DrawerOverlay: An overlay Drawer renders on top of the whole page. By default blocks the screen and will require the user's full attention. 
```
---
- [ ] DrawerInline description

> DrawerInline: Represents an inline Drawer. This is rendered within a container and can be placed next to any content.

```suggestion
 DrawerInline: An inline Drawe renders within a container and can be placed next to any content.
```
---
- [ ] Drawer Description

> Drawer: It is a combination of DrawerOverlay and DrawerInline. Used when toggling between the two modes is necessary. Often used for responsiveness.

```suggestion
Drawer: A combination of DrawerOverlay and DrawerInline. Used when toggling between the two modes is necessary. Often used for responsiveness.
```

---

- [ ] open and defaultOpen props

Consider adding a note to `open` that is it mutually exclusive with `defaultOpen` and vice versa. See `value` and `defaultValue` for an example of this on [Input](https://react.fluentui.dev/?path=/docs/components-input--default).

---

- [ ] Overlay description

DrawerOverlay contains supplementary content and ~are~ **is** used for complex creation, edit,...

By default, drawer is blocking and signifies that the ~users~ **user's** full attention...

### PR DESCRIPTION
**Pull request by marcosmoura**
## Previous Behavior
Currently the Drawer components are named using `<Scope><Variant>` naming convention (DrawerInline/DrawerOverlay)

## New Behavior
After internal discussions, the components should be named `<Variant><Scope>` (InlineDrawer/OverlayDrawer)

**Note:**
This is technically a breaking change, but as the component is unstable, we can make that happen.

This also fixes #29238

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                   |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `92.961 kB`<br />`28.372 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                             |  `27.197 kB`<br />`8.779 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                               |  `28.098 kB`<br />`9.074 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                             |  `25.603 kB`<br />`9.283 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                          | `40.007 kB`<br />`11.277 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                          | `47.363 kB`<br />`12.768 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                  | `44.781 kB`<br />`12.649 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                | `52.795 kB`<br />`14.236 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                              | `57.054 kB`<br />`13.176 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>          | `89.578 kB`<br />`26.737 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                       |  `21.112 kB`<br />`7.812 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                             |  `22.721 kB`<br />`8.611 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                             |  `16.096 kB`<br />`6.317 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                             |   `14.435 kB`<br />`5.83 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                               |  `17.328 kB`<br />`6.947 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                       | `119.11 kB`<br />`37.616 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                          |  `12.255 kB`<br />`4.504 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                          |  `25.305 kB`<br />`9.162 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                        |  `22.687 kB`<br />`8.399 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                       |  `23.688 kB`<br />`8.536 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                              |  `17.065 kB`<br />`6.645 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                            |  `20.237 kB`<br />`6.965 kB` |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr>                         | `92.478 kB`<br />`27.738 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                       |  `52.322 kB`<br />`18.42 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b03a47260f977d472f34e3402dc89d3e3d56f760'>b03a47260f977d472f34e3402dc89d3e3d56f760</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Avatar-mount_base.html">651</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Avatar-mount.html">601</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Button-mount_base.html">320</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Button-mount.html">301</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Field-mount_base.html">1123</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Field-mount.html">1070</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProvider-mount_base.html">712</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProvider-mount.html">716</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-mount_base.html">83</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-mount.html">85</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">62</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/InfoButton-mount_base.html">16</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/InfoButton-mount.html">8</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/MakeStyles-mount_base.html">826</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/MakeStyles-mount.html">853</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Persona-mount_base.html">1755</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/Persona-mount.html">1645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/SpinButton-mount_base.html">1367</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29523/perf-test-react-components/SpinButton-mount.html">1316</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts [hunk 1]
```diff
@@ -90,8 +90,8 @@ export type CustomStyleHooksContextValue = Partial<{
   useDataGridHeaderCellStyles_unstable: CustomStyleHook;
   useDataGridSelectionCellStyles_unstable: CustomStyleHook;
   useDrawerStyles_unstable: CustomStyleHook;
-  useDrawerInlineStyles_unstable: CustomStyleHook;
-  useDrawerOverlayStyles_unstable: CustomStyleHook;
```
Comment (Hotell) [inline packages/react-components/react-shared-contexts/src/CustomStyleHooksContext/CustomStyleHooksContext.ts:94]: discussed offline with @marcosmoura 

we need to keep both old and new hooks until v10.

**Why:**

- this package is already stable, removing things from public api === breaking change
- this will need to live here even after card is released as stable
- card needs to maintain proper order of calling these hooks ( `useDrawerOverlayStyles_unstable` then ` useOverlayDrawerStyles_unstable` )

In general this is not the best architectural decision as we create **indirect tight coupling** that cannot be checked if it works ( we need to provide integration tests to this package ).

This issues can happen also with our new release strategy concept for unstable packages (`*-preview`)

**Action points:**

- create RFC describing the issue
- present at tech sync

File: packages/react-components/react-components/CHANGELOG.md [hunk 2]
```diff
@@ -138,7 +138,7 @@ Thu, 12 Oct 2023 14:55:42 GMT
 
 ## [9.34.3](https://github.com/microsoft/fluentui/tree/@fluentui/react-components_v9.34.3)
 
-Wed, 11 Oct 2023 13:54:22 GMT
```
Comment (Hotell) [inline packages/react-components/react-components/CHANGELOG.md:141]: huh, how did this happen ? can you please undo these changes ?

File: packages/react-components/react-drawer/CHANGELOG.md [hunk 3]
```diff
@@ -54,7 +54,7 @@ Wed, 11 Oct 2023 13:54:26 GMT
 
 ## [9.0.0-beta.33](https://github.com/microsoft/fluentui/tree/@fluentui/react-drawer_v9.0.0-beta.33)
 
-Mon, 09 Oct 2023 20:45:41 GMT 
+Mon, 09 Oct 2023 20:45:41 GMT
```
Comment (Hotell) [inline packages/react-components/react-drawer/CHANGELOG.md:57]: https://github.com/microsoft/fluentui/pull/29523/files#r1371732769

File: packages/react-components/react-components/CHANGELOG.json [hunk 4]
```diff
@@ -1322,7 +1322,7 @@
             "author": "marcosvmmoura@gmail.com",
             "package": "@fluentui/react-drawer",
             "commit": "fbaba50e246b708d0fcac7d79d0f99df5228ce15",
-            "comment": "fix: remove defaultOpen prop from DrawerInline as it had no effect"
```
Comment (Hotell) [inline packages/react-components/react-components/CHANGELOG.json:1325]: dont change changelog manually, at this stage. it needs to be identical to the release version at that time.

File: change/@fluentui-react-drawer-a264b8f6-8d3e-44c6-9c17-b5ace7f83e38.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "refactor: rename main DrawerOverlay/DrawerInline components",
```
Comment (Hotell) [inline change/@fluentui-react-drawer-a264b8f6-8d3e-44c6-9c17-b5ace7f83e38.json:3]: this is breaking change, lets make it obvious

in react-components/unstable its not as we are doing the rename re-exports

```suggestion
  "comment": "BREAKING CHANGE: rename main DrawerOverlay/DrawerInline components",
```

File: change/@fluentui-react-provider-8a587528-2d75-439f-9301-2505033412fb.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "refactor: rename main DrawerOverlay/DrawerInline components",
```
Comment (Hotell) [inline change/@fluentui-react-provider-8a587528-2d75-439f-9301-2505033412fb.json:3]: can we improve this ? its a bit confusing from changelog POV what change does this mean.

```suggestion
  "comment": "feat: add new drawer global style hooks that are needed by drawer public api changes",
```

**Review by bsunderhus — APPROVED** [review]

**Review by Hotell — APPROVED** [review]
